vizro-core/src/vizro/models/_components/form/range_slider.py

```diff
@@ -40,7 +40,7 @@ class RangeSlider(VizroBaseModel):
     min: Optional[float] = Field(None, description="Start value for slider.")
     max: Optional[float] = Field(None, description="End value for slider.")
     step: Optional[float] = Field(None, description="Step-size for marks on slider.")
-    marks: Optional[Dict[float, str]] = Field({}, description="Marks to be displayed on slider.")
+    marks: Optional[Dict[int, Union[str, Dict[str, str]]]] = Field({}, description="Marks to be displayed on slider.")
```

Comment:
We should probably go with the following type:
`marks: Optional[Dict[int, Union[str, Dict[str, Any]]]]`